### PR TITLE
New post "Migrating kafka with MirrorMaker2"

### PR DESCRIPTION
**Pull request by rmarting**
Proposal for a new blog post "Migrating Kafka clusters with MirrorMaker2 and Strimzi".

Content based in my own personal [blog post](https://blog.jromanmartin.io/2021/11/19/migrating-kafka-with-mirror-maker2.html) and ready for the review of the Strimzi community.

Authors: Roman Martin (rmarting), and Manuel Schindler (maschind)

### CODE REVIEW

**Comment by maschind** [comment]
I hope the strimzi community will like it. Very relevant and I used the scenario already in the field!

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (scholzj) [inline _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:174]: Maybe we can talk about Strimzi here instead of `Red Hat AMQ Streams`.

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (scholzj) [inline _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:182]: Again, using Strimzi would be probably the proper way here.

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. This looks real useful. I left few comments. We will also (on the Strimzi side) need to look into how we can fix the multiple authors issue.

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (rmarting) [reply _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:5]: Will it be a blocker to publish the blog? If so, we could remove one of the authors (e.g. myself) when you accepted this blog.

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (rmarting) [reply _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:174]: Sure! It was the original approach, but ... I forgot these references. :thinking:

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (rmarting) [reply _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:182]: Idem as above! Fixing!

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (scholzj) [reply _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:5]: I'm sure it is not a blocker from our side. It is more that we want to give you the credit ;-)

If you would be fine with it, maybe the easiest way would be that instead of having two authors to have just one author with two names. Something like this:
```yaml
roman_martin_manuel_schindler:
  name: Roman Martin, Manuel Schindler
```

That would get both names there, just no link.

Comment (rmarting) [reply _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:5]: I liked your approach! Fixed in that way!

**Review by scholzj — APPROVED** [review]
Nice post. Thanks.

**Comment by rmarting** [comment]
Thank you so much to accept us our small contribution to the Strimzi Blog!!!

**Comment by scholzj** [comment]
Just FYI: It still needs the approval of others.

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (ppatierno) [inline _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:123]: "based on" ?

File: _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md
Comment (ppatierno) [inline _posts/2021-11-19-migrating-kafka-with-mirror-maker2.md:138]: we are talking about detection of configuration changes on brokers right? should be clearer here?

**Review by ppatierno — APPROVED** [review]
First of all thanks for the contribution! very nice article!
I am approving it but I left a couple of nits and I have another point to raise: I think that the article could talk about Kubernetes in general, because it's going to work on it, not needed to focus on OpenShift as it was a pre-requisite.
Maybe it should refer to Kubernetes but then explain that your specific use case was on OpenShift which is an enterprise distribution of Kubernetes by Red Hat. 
At least this is my view, maybe other maintainers could have a different one.

**Comment by scholzj** [comment]
> I am approving it but I left a couple of nits and I have another point to raise: I think that the article could talk about Kubernetes in general, because it's going to work on it, not needed to focus on OpenShift as it was a pre-requisite.
Maybe it should refer to Kubernetes but then explain that your specific use case was on OpenShift which is an enterprise distribution of Kubernetes by Red Hat.
> At least this is my view, maybe other maintainers could have a different one.

I don't know. I was thinking about that part as well. The AMQ Streams references didn't seemed suitable on a Strimzi blog since Strimzi should be the focus. But OpenShift is one of the platforms we support and which many users use. So that seems fine to me. That said, I think it might be worth mentioning at the end or in the beginning that this should work on other Kube distros as well.

**Comment by ppatierno** [comment]
I didn't read about "AMQ Streams" on OpenShift but more "Strimzi" on OpenShift but now I noticed that the repo with the demo is named as AMQ Streams even if it refers to Strimzi AFAIU.

**Comment by scholzj** [comment]
>I didn't read about "AMQ Streams" on OpenShift but more "Strimzi" on OpenShift but now I noticed that the repo with the demo is named as AMQ Streams even if it refers to Strimzi AFAIU.

It is just a repo name, nothing else.

**Comment by ppatierno** [comment]
> It is just a repo name, nothing else.

Yeah exactly it's totally fine with me even because anyway it talks about Strimzi in the end.
I would just change the article here to talk more about Kubernetes or as you said making clear that the use case was on OpenShift but everything would work with Kubernetes. Anyway the things to change from examples would be the external listeners not using "route".

**Comment by rmarting** [comment]
Thanks @ppatierno and @scholzj for your feedback and comments.

I agree that this article should be aligned with the upstream community of Strimzi, so:

- Removed any reference about Red Hat AMQ Streams
- Added a comment to clarify that the article is based in OpenShift, but it works in Kubernetes too
- Renamed the sample repo to strimzi-migration-demo (main branch for Strimzi and a rh-amq-streams branch for the product)

I hope these changes helped for your review, and approvement.

**Comment by ppatierno** [comment]
@rmarting The changes look good to me. If @scholzj agrees we could publish this article tomorrow, but it means you have to change the date (which currently at 2021-11-09).

**Comment by scholzj** [comment]
The date can be fixed while merging it, no need to worry about it. Tomorrow sounds good.

**Review by Frawless — APPROVED** [review]

**Comment by scholzj** [comment]
Thanks for the blog post @rmarting & @maschind

**Comment by rmarting** [comment]
Thanks Strimzi Community to accept our small contribution!!!

**Pull request merged**
